src/app/static/src/app/store/load/mutations.ts

```diff
@@ -27,5 +36,38 @@ export const mutations: MutationTree<LoadState> & LoadMutations = {
         //For both load success and clear error
         state.loadingState = LoadingState.NotLoading;
         state.loadError = null;
+    },
+    RehydrateResult(state: LoadState, action: PayloadWithType<ProjectRehydrateResultResponse>) {
+        state.rehydrateResult = action.payload;
+        state.preparing = false;
+        state.loadingState = LoadingState.NotLoading;
+    },
+    RehydrateResultError(state: LoadState, action: PayloadWithType<Error>) {
+        state.loadError = action.payload;
+        state.preparing = false;
+        state.loadingState = LoadingState.LoadFailed;
+        if (state.statusPollId > -1) {
+            stopPolling(state);
+        }
+    },
+    PreparingModelOutput(state: LoadState, action: PayloadWithType<ProjectRehydrateSubmitResponse>) {
+        state.downloadId = action.payload.id;
```

Comment:
Can we call this `rehydrateId`? Download Id is used when generating the downloadable files.